cmd/gateway/s3/gateway-s3.go

```diff
@@ -313,8 +317,9 @@ func (l *s3Objects) GetObject(ctx context.Context, bucket string, key string, st
 	}
 
 	opts := miniogo.GetObjectOptions{}
-	opts.ServerSideEncryption = o.ServerSideEncryption
-
+	if o.ServerSideEncryption != nil {
```

Comment:
`nil` check not required - if not set `opts.ServerSideEncryption` is `nil` anyway.